File: src/main/java/cat/xlagunas/andrtc/token/TokenRepository.java
package cat.xlagunas.andrtc.token;

import java.util.List;

public interface TokenRepository {

    long addToken(long userId, String token, String platform) throws ExistingTokenException;

    List<Token> getUserTokens(long userId);

    List<Token> getUserListToken(List<Long> userId);

    boolean removeToken(long userId, String token);
}

File: src/main/java/cat/xlagunas/andrtc/token/TokenRepository.kt
package cat.xlagunas.andrtc.token

interface TokenRepository {

    @Throws(ExistingTokenException::class)
    fun addToken(userId: Long, token: String, platform: String): Long

    fun getUserTokens(userId: Long): List<Token>

    fun getUserListToken(userId: List<Long>): List<Token>

    fun removeToken(userId: Long, token: String): Boolean
}
